pytorch_lightning/callbacks/base.py

```diff
@@ -33,6 +33,14 @@ class Callback(abc.ABC):
     Subclass this class and override any of the relevant hooks
     """
 
+    @property
+    def state_id(self) -> str:
+        return self.__class__.__qualname__
+
+    @property
+    def _legacy_state_id(self) -> Type:
+        return type(self)
+
```

Comment:
nit: add docstrings to describe these? otherwise the historical context for the legacy state id will quickly be lost

how long should we support the legacy state id? as long as possible? for X number of PyTorch Lightning versions? should we expect users to use checkpoint upgrade utilities to keep these compliant?

Reply:
added docs

I would prefer as long as possible since we can upgrade the checkpoint dict automatically when it gets loaded. See #8558. Do you have any suggestions of how long?

Reply:
> as long as possible?

I think so, breaking old checkpoints is not cool. Especially if we value reproducibility.